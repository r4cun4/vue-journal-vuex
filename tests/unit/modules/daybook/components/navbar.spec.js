import { shallowMount } from "@vue/test-utils"

import TheNavbar from '@/modules/daybook/components/TheNavbar'
import createVuexStore from '../../../mock-data/mock-store'


describe('Pruebas en el TheNavbar component', () => {

    const store = createVuexStore({
        user: {
            name: 'Juan Carlos',
            email: 'juan@gmail.com'
        },
        status: 'authenticated',
        idToken: 'ABC',
        refreshToken: 'XYZ'
    })

    beforeEach(() => jest.clearAllMocks())

    test('Debe de mostrar el componente correctamente', () => {

        const wrapper = shallowMount( TheNavbar, {
            global: {
                plugins: [ store ]
            }
        })

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('click en el logout, debe de cerrar sesiÃ³n y redireccionar', async() => {

        const wrapper = shallowMount( TheNavbar, {
            global: {
                plugins: [ store ]
            }
        })

        await wrapper.find('button').trigger('click')

        expect( wrapper.router.push ).toHaveBeenCalledWith({ name: 'login' })
        expect( store.state.auth ).toEqual({
            user: null,
            status: 'not-authenticated',
            idToken: null,
            refreshToken: null
        })

    })

})