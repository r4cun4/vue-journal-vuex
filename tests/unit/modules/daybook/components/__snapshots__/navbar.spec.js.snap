// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el TheNavbar component Debe de mostrar el componente correctamente 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white"><img src="" alt="Vue Logo" height="24" class="d-inline-block align-text-top mx-2"> Juan Carlos</a>
  <div class="d-flex"><button class="btn btn-outline-info"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
